src/languages/php.js

```diff
@@ -11,23 +11,105 @@ Category: common
  * @returns {LanguageDetail}
  * */
 export default function(hljs) {
+  const regex = hljs.regex;
+  const LABEL = '[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*' +
+    // negative look-ahead tries to avoid matching patterns that are not
+    // Perl at all like $ident$, @ident@, etc.
+    '(?![A-Za-z0-9])(?![$])';
+  const PASCAL_CASE_CLASS_NAME = regex.concat("[A-Z]", LABEL);
   const VARIABLE = {
-    className: 'variable',
-    begin: '\\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*' +
-      // negative look-ahead tries to avoid matching patterns that are not
-      // Perl at all like $ident$, @ident@, etc.
-      `(?![A-Za-z0-9])(?![$])`
+    scope: 'variable',
+    match: '\\$+' + LABEL,
+  };
+  const CONSTANT_CALL = regex.concat(LABEL, "\\b(?!\\()");
+  const CONSTANT = {
+    variants: [
+      {
+        match: [
```

Comment:
As this is a declaration rather than a reference I think it likely needs it's own rule rather than being mixed into this huge `variants`.

Reply:
Can you explain it more or add some link to the code/example?

Reply:
I just mean it doesn't belong with all the rest, make a singular rule:

```
const CONSTANT_DEFINITION = {
   match: [
           /const/,
           /\s+/,
           LABEL,
           /\s*=/,
}
```

Reply:
All the rest deal with `x::y`, so it make sense to group them.